src/hotspot/cpu/riscv/sharedRuntime_riscv.cpp

```diff
@@ -905,6 +905,48 @@ static void continuation_enter_cleanup(MacroAssembler* masm) {
 
   __ ld(t0, Address(sp, ContinuationEntry::parent_cont_fastpath_offset()));
   __ sd(t0, Address(xthread, JavaThread::cont_fastpath_offset()));
+
+  if (CheckJNICalls) {
+    // Check if this is a virtual thread continuation
+    Label L_skip_vthread_code;
+    __ lw(t0, Address(sp, ContinuationEntry::flags_offset()));
```

Comment:
@fbredber : Nit: maybe it's better to use `lwu` here instead of `lw` to load the flags? `lw` would do sign-extension for the upper 32 bits which I don't think is wanted in case when we have more meaningful bits in flags. While `lwu` zeros the upper 32 bits.

Reply:
Good catch, thanks!

Reply:
Thanks for the suggestion @RealFYang I have changed it.